crates/driver/src/print_build.rs

```diff
@@ -474,6 +474,10 @@ where
     }
 
     apply_attr_string_lists(&mut extra_kv_pairs, &graph_node.node_metadata);
+    let deps = extra_kv_pairs
```

Comment:
can we comment why we are taking this here? Is it because extra_kv_pairs is being mutated below?

I am having a hard time following this function since it is very large IIUC and has a lot of nested internal functions.

Reply:
Addressed in c97d4b7c8d900f53d63d9e54182d7ffbc0fc7945

    // before we give extra_kv_pairs away to make the main target,
    // we need to clone deps here for a later use in secondaries.